src/NexusMods.App.UI/Pages/CollectionDownload/CollectionDownloadViewModel.cs

```diff
@@ -89,8 +89,18 @@ public CollectionDownloadViewModel(
         CommandDownloadRequiredItems = _isDownloadingRequiredItems.CombineLatest(_canDownloadRequiredItems, static (isDownloading, canDownload) => !isDownloading && canDownload).ToReactiveCommand<Unit>(
             executeAsync: async (_, cancellationToken) =>
             {
-                if (loginManager.IsPremium) await collectionDownloader.DownloadItems(_revision, itemType: CollectionDownloader.ItemType.Required, db: connection.Db, cancellationToken: cancellationToken);
-                else overlayController.Enqueue(serviceProvider.GetRequiredService<IUpgradeToPremiumViewModel>());
+                if (!loginManager.IsLoggedIn)
+                {
+                    await loginManager.LoginAsync(cancellationToken);
+                    if (!await loginManager.GetIsUserLoggedInAsync(cancellationToken)) return;
+                }
+                if (!loginManager.IsPremium)
+                {
+                    overlayController.Enqueue(serviceProvider.GetRequiredService<IUpgradeToPremiumViewModel>());
+                    return;
+                }
+   
+                await collectionDownloader.DownloadItems(_revision, itemType: CollectionDownloader.ItemType.Required, db: connection.Db, cancellationToken: cancellationToken);
```

Comment:
This looks weird, this code shouldn't try to login the user. If we have an issue with the login status not propagating properly, then this code won't fix that issue.